esx_service/utils/auth_data_test.py

```diff
@@ -0,0 +1,591 @@
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+# Tests for auth.py
+
+import unittest
+import os
+import os.path
+import auth_data
+import uuid
+import auth_data_const
+
+class TestAuthDataModel(unittest.TestCase):
+    """
+    Test the Authorization data model via the AuthorizationDataManager
+
+    """
+    db_path = "/tmp/test-auth.db"
+    
+    def setUp(self):
+        """ Create the auth DB and connect to the DB for each test """
+
+        try:
+            os.unlink(self.db_path)
+        except:
+            pass
+            
+        self.auth_mgr = auth_data.AuthorizationDataManager(self.db_path)
+        self.auth_mgr.connect()
+        #self.auth_mgr.create_tables()
+
+    def tearDown(self):
+        """ Tear down the auth DB after each test """
+
+        os.unlink(self.db_path)
+
+    def get_privileges(self):
+        privileges = [{'datastore': 'datastore1',
+                     'global_visibility': 0,
+                      'create_volume': 0,
+                      'delete_volume': 0,
+                     'mount_volume': 0,
+                      'max_volume_size': 0,
+                      'usage_quota': 0}]
+        return privileges
+    
+    def get_default_datastore_and_privileges(self):
+        default_datastore = 'default_ds'
+        default_privileges = [{'datastore': default_datastore,
+                              'global_visibility': 0,
+                              'create_volume': 0,
+                              'delete_volume': 0,
+                              'mount_volume': 0,
+                              'max_volume_size': 0,
+                              'usage_quota': 0}]
+        return default_datastore, default_privileges
+    
+    def get_row_from_tenants_table(self, tenant_uuid):
```

Comment:
should not get_rows_from* be functions in the auth package, just called from here?